fighters/mario/src/opff.rs

```diff
@@ -131,12 +131,99 @@ unsafe fn double_fireball(fighter: &mut L2CFighterCommon, boma: &mut BattleObjec
+                    smash::app::lua_bind::FighterKineticEnergyGravity::set_stable_speed(fighter_gravity, -0.2);
+                    KineticModule::mul_speed(boma, &air_fri, *FIGHTER_KINETIC_ENERGY_ID_GRAVITY);
+                }
+                if frame >= 50.0 {
+                    VarModule::off_flag(fighter.battle_object, vars::mario::instance::DISABLE_DSPECIAL_STALL);
+                }
+            } 
+        }
+    }
+    if fighter.is_situation(*SITUATION_KIND_GROUND) || fighter.is_situation(*SITUATION_KIND_CLIFF) || fighter.is_status(*FIGHTER_STATUS_KIND_REBIRTH) || fighter.is_situation(*SITUATION_KIND_LADDER) {
+            VarModule::on_flag(fighter.battle_object, vars::mario::instance::DISABLE_DSPECIAL_STALL);
+    }
+}
+
+// Grounded SMG spin movement
+unsafe fn galaxy_spin_move(fighter: &mut L2CFighterCommon, boma: &mut BattleObjectModuleAccessor, status_kind: i32, motion_kind: u64, situation_kind: i32, frame: f32, stick_x: f32, facing: f32) {
+    if motion_kind == hash40("special_lw_light") {
+        let current_speed =  KineticModule::get_sum_speed_x(boma, *KINETIC_ENERGY_RESERVE_ATTRIBUTE_MAIN);
+        if fighter.is_status(*FIGHTER_MARIO_STATUS_KIND_SPECIAL_LW_SHOOT) && fighter.is_situation(*SITUATION_KIND_GROUND){
+            if frame >= 6.0 && frame < 20.0 { 
+                VarModule::off_flag(fighter.battle_object, vars::mario::instance::DISABLE_DSPECIAL_STALL);
+                if stick_x != 0.0 {
+                    let motion_vec = x_motion_vec(1.2, stick_x);
+                    KineticModule::add_speed_outside(boma, *KINETIC_OUTSIDE_ENERGY_TYPE_WIND_NO_ADDITION, &motion_vec);
+                }
+            }
+            else if frame >= 20.0 && frame < 30.0 {
+                if stick_x != 0.0 {
+                    let motion_vec = x_motion_vec(1.0, stick_x);
+                    KineticModule::add_speed_outside(boma, *KINETIC_OUTSIDE_ENERGY_TYPE_WIND_NO_ADDITION, &motion_vec);
+                }
+            }
+            else if frame >= 30.0 && frame < 40.0 {
+                if stick_x != 0.0 {
+                    let motion_vec = x_motion_vec(0.8, stick_x);
+                    KineticModule::add_speed_outside(boma, *KINETIC_OUTSIDE_ENERGY_TYPE_WIND_NO_ADDITION, &motion_vec);
+                }
+            }
+            else if frame >= 40.0 && frame < 50.0 {
+                if stick_x != 0.0 {
+               
+                    let motion_vec = x_motion_vec(0.5, stick_x);
+                    KineticModule::add_speed_outside(boma, *KINETIC_OUTSIDE_ENERGY_TYPE_WIND_NO_ADDITION, &motion_vec);
+                }
+            }
+            else if frame >= 50.0 && frame < 55.0 {
+                if stick_x != 0.0 {
+                    let motion_vec = x_motion_vec(0.2, stick_x);
+                    KineticModule::add_speed_outside(boma, *KINETIC_OUTSIDE_ENERGY_TYPE_WIND_NO_ADDITION, &motion_vec);
+                }
```

Comment:
Where can I find an example of linear interp?

Reply:
That and the reason is that I didn't know there was a simpler/cleaner answer. I did this because this was the best that I can do with my knowledge of either rust or smashline.